internal/node-patches/package.json

```diff
@@ -0,0 +1,29 @@
+{
+  "name": "@bazel/node-patches",
```

Comment:
why is this under `/internal`?

We have a convention about each directory under `packages` getting published to npm so I would have expected it to go there

Reply:
we moved it last minute friday because @gregmagolan felt it fit a little better in internal.
it doesn't fit exactly anywhere because it's developed like a nodejs project and the chicken/egg of bootstrapping node in bazel

Reply:
in `publish_release.sh` we have
```
readonly NPM_PACKAGE_LABELS=`$BAZEL query --output=label 'kind("npm_package rule", //packages/...) - attr("tags", "\[.*do-not-publish.*\]", //packages/...)'`
```

this is because the `/internal` directory is meant to be distributed with the built-in tgz, while things in `packages/*` are published to npm. I would rather preserve this convention rather than need special cases in the publish_release.sh and likely some other places.